dist/index.js

```diff
@@ -49801,12 +49832,23 @@ async function waitForInstance(taskId, taskName) {
 
 async function terminateInstanceRequest(runnerName) {
   const url = config.input.slabUrl
-  const payload = {
-    region: config.input.region,
-    runner_name: runnerName,
-    action: 'terminate',
-    sha: config.githubContext.sha
+  let payload
+  if (config.input.profile) {
+    payload = {
+      details: {
+        profile: config.input.profile
+      }
+    }
+  } else {
+    payload = {
+      details: { custom_stop: { region: config.input.region } }
+    }
   }
+  payload.runner_name = runnerName
+  payload.action = 'terminate'
+  payload.sha = config.githubContext.sha
+  payload.git_ref = config.githubContext.ref
```

Comment:
is it possible to have the const payload as before provided we fill details ahead of time or not ?

like

```js
let details

  if (config.input.profile) {
    detail= {
      profile: config.input.profile
    }
  } else {
      details = { custom_stop: { region: config.input.region } }
  }

  const payload = {
    "details": details,
    "runner_name": ...
  }
```

?